pagedEmbed response. initial commit

Therosin opened the pull request.

Therosin committed:
pagedEmbed response. initial commit

Therosin committed:
fixed invalid shorthand property

Therosin committed:
cleanup

Therosin committed:
silly mistakes.. forgot to remove the checks for message_text as were not passing that around anymore, _should pay more attention._

Therosin committed:
more mistakes _corrected invalid reference to guilds left over from old code implementation_

Therosin committed:
add some logging/pass proper server ref

Therosin committed:
no message

Therosin committed:
Updated.
MessageButton not supported in discord.js < 13
changed to a new simplified method, currently only shows 1 player per page

Therosin committed:
forgot to remove the requires line for removed pagedEmbed class

Therosin committed:
UPDATED: Compleated ./utils/pQueue modulel

Therosin committed:
update: pageEmbed Response, cleaned up responses. and grouped them into pages of 2 for now while testing.

Therosin committed:
BUGFIX: missing reference

Therosin committed:
FIX: undefined var

Therosin committed:
UPDATED: cleaned up server-list command

Therosin committed:
UPDATE: remove playerList from server-status command + general cleanup

Therosin committed:
UPDATE: changed banlist-show command to use sendPagedEmbed

Therosin committed:
CHORE: corrected missused variable assignment in slicedArray()

Therosin committed:
UPDATE: cleaned up banlist-show command

Therosin committed:
UPDATE: changed whitelist-show to use sendPagedEmbed

Therosin committed:
UPDATE: cleaned up responses for server-list/banlist-show/whitelist-show commands

Therosin committed:
FIXED: forgot to push playerDetail to PlayerList array

Therosin committed:
fixed a silly mistake forgeting to join the pages before sendPagedEmbed

Therosin committed:
FIX: fixed missing `this` reference in forEach loops

Therosin committed:
no message

Therosin committed:
reverted to a basic loop try fixed the undefined reference in returned response

Therosin committed:
FIX: broke everything.. forgot to make the promise async
(realy have to sort out a better way to test localy. without creating a 3rd test bot)

Therosin reviewed (commented):
These files are still a mess of spaghetti-code. but for function as intended.
we will need to go through and refactor these soon.

- server-info is basicaly a duplicate of server-status at this point. mabe we need to find more server details to show admins. or just figure out what needs to be shown for all... :shrug: not sure. seems a waste to have both command still is all.

Amoebawp commented:
- Will look at the files asap.

As for server-info/server-status :
The main difference is the authorization ; 
info has more details than status, but status can be called by any user on the guild. User doesn't need to have the miscord-user role.
It's also designed to be more readable and accessible.
However, since you got a working player list now, we could re-integrate it under info.

Also, if possible, idea would be to consider info as a main "hub" for commands : give clickable options, either in the embed directly, or reactions, to trigger other commands (chat bubble icon > server-message / computer icon > server-restart / etc etc).

-- edit add : maybe if we can create other lightened version of commands, to be accessible by any users, it would be wise to separate them from server-* group, to have a group on their own, like a "Public" group or something. Ideally, letting the admin choose individually which commands to allow/disallow for that public group would be purrfect.

Amoebawp committed:
Minor visual adjustments

Therosin committed:
local merge from pitiviers changes

Therosin committed:
fix package.json

Therosin committed:
fixed merge

Therosin committed:
FIX: apiserver. corrected merge
--- this should fix the merge / stash issue from erlier. annd bring everything back up to scratch...
this commit also changes a few dependant libraries so if running a local copy you need to `yarn clean; yarn install` again to pull all deps

Amoebawp committed:
Minor visual changes

Amoebawp committed:
Fix missing \n
